private/bufpkg/bufconfig/write.go

```diff
@@ -273,30 +257,173 @@ func writeConfig(
+	}
+	// Write out raw default config without comments using externalConfig{V1, V1Beta1}
+	return writeExternalConfig(ctx, writeBucket, version, config, dependencies)
+}
+
+func writeExternalConfig(
+	ctx context.Context,
+	writeBucket storage.WriteBucket,
+	version string,
+	config *Config,
+	dependencies []string,
+) error {
+	var name string
+	if config.ModuleIdentity != nil {
+		name = config.ModuleIdentity.IdentityString()
+	}
+	breakingConfig := config.Breaking
+	lintConfig := config.Lint
+	switch version {
+	case V1Beta1Version:
+		var externalBreakingConfig bufbreakingconfig.ExternalConfigV1Beta1
+		if breakingConfig != nil {
+			externalBreakingConfig = bufbreakingconfig.ExternalConfigV1Beta1ForConfig(breakingConfig)
+		}
+		var externalLintConfig buflintconfig.ExternalConfigV1Beta1
+		if lintConfig != nil {
+			externalLintConfig = buflintconfig.ExternalConfigV1Beta1ForConfig(lintConfig)
+		}
+		externalConfig := ExternalConfigV1Beta1{
+			Name:     name,
+			Version:  config.Version,
+			Deps:     dependencies,
+			Breaking: externalBreakingConfig,
+			Lint:     externalLintConfig,
+		}
+		var buffer bytes.Buffer
+		encoder := yaml.NewEncoder(&buffer)
+		encoder.SetIndent(2)
+		if err := encoder.Encode(externalConfig); err != nil {
+			return err
+		}
+		return storage.PutPath(ctx, writeBucket, ExternalConfigV1Beta1FilePath, buffer.Bytes())
+	case V1Version:
+		var externalBreakingConfig bufbreakingconfig.ExternalConfigV1
+		if breakingConfig != nil {
+			externalBreakingConfig = bufbreakingconfig.ExternalConfigV1ForConfig(breakingConfig)
+		}
+		var externalLintConfig buflintconfig.ExternalConfigV1
+		if lintConfig != nil {
+			externalLintConfig = buflintconfig.ExternalConfigV1ForConfig(lintConfig)
+		}
+		externalConfig := ExternalConfigV1{
+			Name:     name,
+			Version:  config.Version,
+			Deps:     dependencies,
+			Breaking: externalBreakingConfig,
+			Lint:     externalLintConfig,
+		}
+		var buffer bytes.Buffer
+		encoder := yaml.NewEncoder(&buffer)
+		encoder.SetIndent(2)
+		if err := encoder.Encode(externalConfig); err != nil {
+			return err
+		}
+		return storage.PutPath(ctx, writeBucket, ExternalConfigV1FilePath, buffer.Bytes())
+	default:
+		return fmt.Errorf(`%s has an invalid "version %s"`, name, version)
+	}
+}
+
+type tmplParam struct {
+	Uncomment      bool
+	Version        string
+	Name           string
+	NameUnset      bool
+	Deps           []string
+	DepsUnset      bool
+	LintConfig     *buflintconfig.Config
+	BreakingConfig *bufbreakingconfig.Config
+}
+
+func newTmplParam(config *Config, dependencies []string, uncomment bool) *tmplParam {
+	var name string
+	var nameUnset bool
+	if config.ModuleIdentity != nil {
+		name = config.ModuleIdentity.IdentityString()
+	}
+	if name == "" {
+		name = exampleName
+		nameUnset = true
 	}
+	var dependenciesUnset bool
+	if len(dependencies) == 0 {
+		dependencies = []string{
+			"buf.build/acme/petapis",
```

Comment:
Why were these removed from being consts?

Reply:
I moved them here since they are only being used for the template rendering, which is only used by `configinit` and I figured we should reduce the number globals. But I can move them back if that is preferred.